### DISCUSSION TITLE
Gitbook Lens - Provide more source data

### DISCUSSION THREAD

**Discussion by dgafka** [GitbookIO/community#254] [Feature Requests]
Hey,
Firstly I need to say big thanks to the team for creating Gitbook Lens. 
I was looking for AI integration to provide answers based on my content and lucky me I found, that Gitbook already provides that :) 
I believe this will be game changer for how we interact with the documentation and how we learn new things. 

Back to the point. I've documentation for [Open-Source PHP Framework Ecotone](https://docs.ecotone.tech/) and I've enabled Lens, which works like a charm. However Lens is limited with the data, that it can build answers on. 
Besides the documentation page, I am also hosting [Blog for Ecotone](https://blog.ecotone.tech/) with a lot of articles, containing more details, granular descriptions and code examples than documentation has. The blog could be additional source of the materials for AI model.

I would like to ask, if there is possibility or plans to add custom pages as source of data for AI Model?

Gitbook Lens could give much more meaningful and detailed answers if that would be the case.
Current Ecotone users and new users could get the answers straight away without additional searching, which would be really powerful.

**Comment by addisonschultz** [comment]
Hey @dgafka, thanks for using GitBook with us! Happy to hear that Lens has been useful to you so far. Totally understand your use-case on adding more sources for lens, it's been talked about internally and requested before, so we definitely have it on the radar.

That being said, there might be a few workarounds for you at the moment - while you wouldn't be able to include more sources into the GitBook Lens search, we do provide an [API for retrieving GitBook Lens searches](https://developer.gitbook.com/gitbook-api/reference/search#post-ai-ask-question-on-all-content-accessible-by-the-authenticated-user) - meaning you could integrate Lens into a different product, or search if you wanted.

The alternative, is maybe if there's a way you could include your blogs somewhere in GitBook alongside the docs? Maybe using something like [GitSync](https://docs.gitbook.com/product-tour/git-sync) could be useful in this scenario?

**Reply by dgafka** [reply]
Hey @addisonschultz , 
Thank you for the answer :) 

> blogs somewhere in GitBook alongside the docs
1. Is there a possibility to create Gitbook pages, that are not visible for end users, yet are indexed by Lens? 
If that's the case, then this workaround would work for my case :) 

2. Are the links to Github snippets embed into page also indexed for Lens?
e.g. https://gist.github.com/dgafka/aadc04d39be2480f0b11afb97b195373

**Reply by addisonschultz** [reply]
Hi @dgafka - unfortunately it's not possible for either of your questions :/ 

It's not possible to have a hidden/non-visible pages right now, but something I've raised for us to look into! The snippets are unfortunately not indexed either - mainly because it's embedding the file, vs the code itself - in markdown it's just a link/reference to the original file

**Reply by dgafka** [reply]
Ok, sure. 
Let me know now in that case, when there will be some solution ready for adding more content to Lens :)
